packages/components/src/internal/shared/validators.ts

```diff
@@ -0,0 +1,14 @@
+import BaseValidator from '../../validators/BaseValidator';
+
+export interface ValidateReturnType {
```

Comment:
I think we should avoid naming return types as `...ReturnType` because this decreases flexibility. E.g. it may feel strange later to create something like:
```
const alwaysTrueValidation: ValidateReturnType = {...}
```
I think we should name it e.g. `ValidationResult`:
```suggestion
export interface ValidationResult {
```